api/client/client.go

```diff
@@ -1086,12 +1086,12 @@ func (c *Client) GenerateOpenSSHCert(ctx context.Context, req *proto.OpenSSHCert
 }
 
 // EmitAuditEvent sends an auditable event to the auth server.
-func (c *Client) EmitAuditEvent(ctx context.Context, event events.AuditEvent) error {
+func (c *Client) EmitAuditEvent(ctx events.AuditContext, event events.AuditEvent) error {
 	grpcEvent, err := events.ToOneOf(event)
 	if err != nil {
 		return trace.Wrap(err)
 	}
-	_, err = c.grpc.EmitAuditEvent(ctx, grpcEvent)
+	_, err = c.grpc.EmitAuditEvent(ctx.Context(), grpcEvent)
```

Comment:
Wonder if we can just change this method instead of introducing a new `events.AuditContext` type ? 
 
```suggestion
	_, err = c.grpc.EmitAuditEvent(context.WithoutCancel(ctx), grpcEvent)
```

Reply:
+1000